File: backend/app/schemas/alert.py
"""Alert schemas for API request/response."""

from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import PaginationMeta


class AlertConfigCreate(BaseModel):
    stock_id: int | None = None
    min_strength: str = Field(default="moderate", pattern="^(strong|moderate|weak)$")
    direction_filter: list[str] | None = None
    channel: str = Field(default="both", pattern="^(discord|email|both)$")


class AlertConfigUpdate(BaseModel):
    stock_id: int | None = None
    min_strength: str | None = None
    direction_filter: list[str] | None = None
    channel: str | None = None
    is_active: bool | None = None


class AlertConfigResponse(BaseModel):
    id: int
    user_id: int
    stock_id: int | None
    ticker: str | None = None
    min_strength: str
    direction_filter: list[str] | None
    channel: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class AlertLogResponse(BaseModel):
    id: int
    signal_id: int
    user_id: int
    channel: str
    sent_at: datetime
    success: bool
    error_message: str | None
    ticker: str | None = None
    direction: str | None = None
    strength: str | None = None

    model_config = {"from_attributes": True}


class PaginatedAlertLogs(BaseModel):
    data: list[AlertLogResponse]
    meta: PaginationMeta


class TestAlertRequest(BaseModel):
    channel: str = Field(default="discord", pattern="^(discord|email|both)$")


class TestAlertResponse(BaseModel):
    success: bool
    message: str

File: frontend/src/types/alert.ts
export interface AlertConfig {
  id: number;
  user_id: number;
  stock_id: number | null;
  ticker: string | null;
  min_strength: "strong" | "moderate" | "weak";
  direction_filter: string[] | null;
  channel: "discord" | "email" | "both";
  is_active: boolean;
  created_at: string;
}

export interface AlertLog {
  id: number;
  signal_id: number;
  user_id: number;
  channel: string;
  sent_at: string;
  success: boolean;
  error_message: string | null;
  ticker: string | null;
  direction: string | null;
  strength: string | null;
}
